packages/labs/preact-signals/src/lib/signal-watcher.ts

```diff
@@ -0,0 +1,52 @@
+/**
+ * @license
+ * Copyright 2023 Google LLC
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+import type {ReactiveElement} from 'lit';
+import {effect} from '@preact/signals-core';
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+type ReactiveElementConstructor = new (...args: any[]) => ReactiveElement;
+
+/**
+ * Adds the ability for a LitElement or other ReactiveElement class to
+ * watch for access to Preact signals during the update lifecycle and
+ * trigger a new update when signals values change.
+ */
+export function SignalWatcher<T extends ReactiveElementConstructor>(
+  Base: T
+): T {
+  return class SignalWatcher extends Base {
+    private _disposeEffect?: () => void;
+
+    override performUpdate() {
+      // ReactiveElement.performUpdate() also does this check, so we want to
+      // also bail early so we don't erroneously appear to not depend on any
+      // signals.
+      if (!this.isUpdatePending) {
+        return;
+      }
+      // If we have a previous effect, dispose it
+      this._disposeEffect?.();
+
+      // We create a new effect to capture all signal access within the
+      // performUpdate phase (update, render, updated, etc) of the element.
+      // Q: Do we need to create a new effect each non-Signal-triggered render?
+      // TODO: test various combinations of render triggers:
```

Comment:
This approach will trigger an update on every signal set, which is probably bad. Here is a prototype for using requestUpdate to avoid this: https://lit.dev/playground/#gist=25c994053f83281f6b0f72c287735c79.

```
performUpdate() {
  if (!this.isUpdatePending) {
    return;
  }
  this._disposeEffect?.();
  let performingUpdate = true;
  this._disposeEffect = effect(() => {
    if (performingUpdate) {
      performingUpdate = false;    
      super.performUpdate();
    } else {
      this.requestUpdate();
    }
  });
}
```

Reply:
done